File: UVA11332.cpp
#include <bits/stdc++.h>
using namespace std;
int main()
{
    string num;
    int tmp_num = 0;
    while(cin >> num)
    {
        if(num == "0")
        {
            break;
        }
        while(num.length() > 1)
        {
            tmp_num = 0;
            for(int i = 0;i < num.length();i++)
            {
                tmp_num += num[i] - '0';
            }
            num = to_string(tmp_num);
        }
        cout << num << endl;
    }
    return 0;
}

File: UVA11332.py
while True:
    num = input()
    if num == "0":
        break
    tmp_num = 0
    while(len(num) > 1):
        tmp_num = 0
        for i in num:
            tmp_num += int(i)
        # print(tmp_num)
        num = str(tmp_num)
    print(num)